infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -0,0 +1,588 @@
+
+        this.detectionTasksScheduler = Executors.newScheduledThreadPool(
+                managementServiceCount,
+                new ThreadFactoryBuilder()
+                        .setDaemon(true)
+                        .setNameFormat(serverContext.getThreadPrefix() + "ManagementService")
+                        .build());
+
+        // Creating the detection worker thread pool.
+        // This thread pool is utilized to dispatch detection tasks at regular intervals in the
+        // detectorTaskScheduler.
+        this.detectionTaskWorkers = Executors.newFixedThreadPool(
+                detectionWorkersCount,
+                new ThreadFactoryBuilder()
+                        .setDaemon(true)
+                        .setNameFormat(serverContext.getThreadPrefix() + "DetectionWorker-%d")
+                        .build());
+    }
+
+    private CorfuRuntime getCorfuRuntime() {
+        return runtimeSingletonResource.get();
+    }
+
+    /**
+     * Executes task to run failure and healing detection every poll interval. (Default: 1 sec)
+     */
+    @Override
+    public void startService() {
+        // Trigger sequencer bootstrap on startup.
+        getCorfuRuntime().getLayoutManagementView()
+                .asyncSequencerBootstrap(serverContext.copyManagementLayout(), detectionTaskWorkers);
+
+        detectionTasksScheduler.scheduleAtFixedRate(
+                () -> runSansThrow(this::detectorTaskScheduler),
+                0, POLICY_EXECUTE_INTERVAL, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Checks if this management client is allowed to handle reconfigurations.
+     * - This client is not authorized to trigger reconfigurations if this node is not a part
+     * of the current layout.
+     *
+     * @return True if node is allowed to handle reconfigurations. False otherwise.
+     */
+    private boolean canHandleReconfigurations() {
+
+        // We check for the following condition here: If the node is NOT a part of the
+        // current layout, it should not attempt to change layout.
+        Layout layout = serverContext.getManagementLayout();
+        if (!layout.getAllServers().contains(serverContext.getLocalEndpoint())) {
+            log.debug("This Server is not a part of the active layout. "
+                    + "Aborting reconfiguration handling.");
+            return false;
+        }
+
+        return true;
+    }
+
+
+    /**
+     * Schedules the detection tasks run by detectorTaskScheduler.
+     * It schedules exactly one instance of the following tasks.
+     * - Failure detection tasks.
+     * - Healing detection tasks.
+     */
+    private synchronized void detectorTaskScheduler() {
+
+        CorfuRuntime corfuRuntime = getCorfuRuntime();
+        getCorfuRuntime().invalidateLayout();
+        serverContext.saveManagementLayout(corfuRuntime.getLayoutView().getLayout());
+
+        if (!canHandleReconfigurations()) {
+            return;
+        }
+
+        heartbeatCounter++;
```

Comment:
add comment for the ticker. The importance of this and that it is only updated here.